spot-vaults/contracts/BillBroker.sol

```diff
@@ -446,24 +456,31 @@ contract BillBroker is
     /// @return mintAmt The amount of LP tokens minted.
     /// @return usdAmtIn The usd tokens to be deposited.
     /// @return perpAmtIn The perp tokens to be deposited.
+    /// @return isFirstMint If the pool currently has no deposits.
     function computeMintAmt(
         uint256 usdAmtMax,
         uint256 perpAmtMax
-    ) public view returns (uint256 mintAmt, uint256 usdAmtIn, uint256 perpAmtIn) {
+    )
+        public
+        view
+        returns (uint256 mintAmt, uint256 usdAmtIn, uint256 perpAmtIn, bool isFirstMint)
+    {
+        uint256 totalSupply_ = totalSupply();
+        isFirstMint = (totalSupply_ <= 0);
+
         if (usdAmtMax <= 0 && perpAmtMax <= 0) {
-            return (0, 0, 0);
+            return (0, 0, 0, isFirstMint);
         }
 
-        uint256 totalSupply_ = totalSupply();
         // During the initial deposit we deposit the entire available amounts.
         // The onus is on the depositor to ensure that the value of USD tokens and
         // perp tokens on first deposit are equivalent.
-        if (totalSupply_ <= 0) {
+        if (isFirstMint) {
             usdAmtIn = usdAmtMax;
             perpAmtIn = perpAmtMax;
             mintAmt = (ONE.mulDiv(usdAmtIn, usdUnitAmt) +
                 ONE.mulDiv(perpAmtIn, perpUnitAmt));
-            mintAmt = mintAmt * INITIAL_RATE;
+            mintAmt = (mintAmt * INITIAL_RATE) - MINIMUM_LIQUIDITY;
```

Comment:
This works, but style-wise it seems cleaner to do the subtraction from the mint amount at the same place where the extra mint amount is minted. Colocating the two parts of this logic reduces chance of these two locations getting out of sync with each other.

When I was reading the mint function up above, it seemed like a bug because nothing was subtracted. So I think it's also less surprising to the next code reader.

Reply:
Moved it to the top